integration-tests/baas-test-server/baasaas.ts

```diff
@@ -0,0 +1,225 @@
+};
+
+function assertUserinfo(value: unknown): asserts value is Userinfo {
+  assert(typeof value === "object" && value !== null);
+  assert("id" in value && typeof value.id === "string");
+  assert("type" in value && typeof value.type === "string");
+  assert("custom_data" in value && typeof value.custom_data === "object");
+  assert("data" in value && typeof value.data === "object");
+  assert("identities" in value && Array.isArray(value.identities));
+  for (const identity of value.identities) {
+    assert(typeof identity === "object");
+    assert("id" in identity && typeof identity.id === "string");
+    assert("provider_type" in identity && typeof identity.provider_type === "string");
+  }
+}
+
+export async function userinfo() {
+  const url = new URL("userinfo", BAASAAS_BASE_URL);
+  const headers = createHeaders(true);
+  const response = await fetch(url, { headers });
+  assert(response.ok);
+  const json = await response.json();
+  assertUserinfo(json);
+  return json;
+}
+
+type Image = {
+  buildVariant: string;
+  imageTag: string;
+  /*
+  _id: string;
+  project: string;
+  order: number;
+  versionId: string;
+  taskId: string;
+  execution: number;
+  timestamp: string;
+  branch: string;
+  revision: string;
+  */
+};
+
+function assertImage(value: unknown): asserts value is Image {
+  assert(typeof value === "object" && value !== null);
+  const object = value as Record<string, unknown>;
+  assert(typeof object.buildVariant === "string");
+  assert(typeof object.imageTag === "string");
+}
+
+type Images = {
+  allBranches: string[];
+  images: Record<string, undefined | Image[]>;
+};
+
+function assertImages(value: unknown): asserts value is Images {
+  assert(typeof value === "object" && value !== null);
+  const object = value as Record<string, unknown>;
+  assert(Array.isArray(object.allBranches));
+  const { images } = object;
+  assert(typeof images === "object" && images !== null);
+  for (const available of Object.values(images)) {
+    assert(Array.isArray(available));
+    for (const image of available) {
+      assertImage(image);
+    }
+  }
+}
+
+export async function listImages() {
+  const url = new URL("images", BAASAAS_BASE_URL);
+  const response = await fetch(url, { headers: createHeaders(false) });
+  assert(response.ok);
+  const json = await response.json();
+  assertImages(json);
+  return json;
+}
+
+type ListContainers = ContainerStatus[];
+
+function assertListContainers(value: unknown): asserts value is ListContainers {
+  assert(Array.isArray(value));
+  for (const element of value) {
+    assertContainerStatus(element);
+  }
+}
+
+export async function listContainers(mine = false) {
+  const url = new URL("listContainers", BAASAAS_BASE_URL);
+  if (mine) {
+    url.searchParams.append("mine", "true");
+  }
+  const response = await fetch(url, { headers: createHeaders(true) });
+  assert(response.ok, response.statusText);
+  const json = await response.json();
+  assertListContainers(json);
```

Comment:
If you want you could possibly refactor these four lines to a separate function returning the json, as I think most functions do these (with the differences being the `fetch` options and the assert function called at the end).

Reply:
True.

Reply:
It was a good idea - I did the refactor and added a bit of error handling too 👍